python/ray/train/_internal/storage.py

```diff
@@ -469,6 +474,12 @@ def __init__(
         self._create_validation_file()
         self._check_validation_file()
 
+        # TODO(justinvyu): This directory is shared across workers of the same
+        # trial on the same node. Saved on restore as well.
+        # Need to consider this for cleaning up. Alternative is to keep a separate
+        # directory for each storage context?
+        self._artifact_staging_dir = Path(tempfile.mkdtemp())
```

Comment:
Change 4: have a special temp folder for trial artifacts that's not the current working directory.